src/main/java/org/jasig/portlet/newsreader/mvc/portlet/reader/NewsController.java

```diff
@@ -160,16 +169,25 @@ public ModelAndView fullStory(@RequestParam Long activeFeed, @RequestParam int i
     	List<NewsConfiguration> feeds = new ArrayList<NewsConfiguration>();
         feeds.addAll(set.getNewsConfigurations());
         Collections.sort(feeds);
+        JSONArray jsonFeeds = new JSONArray();
         List<String> knownFeeds = new ArrayList<String>();
         for(NewsConfiguration feed : feeds) {
-           	knownFeeds.add(String.valueOf(feed.getId()));
+            if (feed.isDisplayed()) {
+                JSONObject jsonFeed = new JSONObject();
+                jsonFeed.put("id", feed.getId());
+                jsonFeed.put("name", feed.getNewsDefinition().getName());
+                jsonFeeds.add(jsonFeed);
+                knownFeeds.add(String.valueOf(feed.getId()));
```

Comment:
Previously the feed would be added to knownFeeds whether displayed or not.  The name ("knownFeeds") suggests that maybe non-displayed feeds SHOULD be included... needs checking in.